Source/diablo.cpp

```diff
@@ -295,12 +294,14 @@ int APIENTRY WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLi
 #endif
 			play_movie("gendata\\logo.smk", TRUE);
 
-		char szValueName[] = "Intro";
-		if (!SRegLoadValue("Diablo", szValueName, 0, &nData))
-			nData = 1;
-		if (nData)
-			play_movie("gendata\\diablo1.smk", TRUE);
-		SRegSaveValue("Diablo", szValueName, 0, 0);
+		{

```

Comment:
I guess this block scope was added so that szValueName may be initialized directly? Would it be possible to use `strcpy` instead and declare `szValueName` at the top of the function? If so, `szValueName` should be declared as an array of the appropriate size. `char szValueName[6]`.

**Edit:** also, it seems strange why we can't use "Intro" directly, but as often with the code. Bin perfect before *pretty* code.

Reply:
I'll try poking it a bit and see if I can find another solution.

Reply:
strcpy gets registered as a function call and using the string directly generates a smaller ASM.